x-pack/plugins/apm/server/lib/observability_overview/get_transactions_per_minute.ts

```diff
@@ -40,7 +43,12 @@ export async function getTransactionsPerMinute({
         size: 0,
         query: {
           bool: {
-            filter: rangeQuery(start, end),
+            filter: [
+              ...rangeQuery(start, end),
+              ...getDocumentTypeFilterForAggregatedTransactions(
+                searchAggregatedTransactions
+              ),
```

Comment:
Don't we have a `metricset.name` that we can automatically add, similar to how we filter by processor.event when using the apm event client?

Reply:
Eg changing `getProcessorEventForAggregatedTransactions` to 

```
export function getProcessorEventForAggregatedTransactions(
  searchAggregatedTransactions: boolean
): ProcessorEvent.metric | ProcessorEvent.transaction {
  return searchAggregatedTransactions
    ? MetricsetName.transactionMetric
    : ProcessorEvent.transaction;
}
```

... it probably shouldn't be called `getProcessorEventForAggregatedTransactions` anymore but `getDocumentTypeForAggregatedTransactions` where `type DocumentType = MetricsetName | ProcessorEvent` (pseudocode)

Reply:
This means, we shouldn't use 

```ts
const params = {
    apm: {
      events: [
        getApmEventForAggregatedTransactions(
          searchAggregatedTransactions
        )
      ],
    },
    body: {},
}

apmEventClient.search(operationName, params)
```

Reply:
Opened https://github.com/elastic/kibana/issues/105713 to track this.